tests/unit/modules/test_mutator.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import pytest
+
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    errors, utils,
+)
+from ansible_collections.sensu.sensu_go.plugins.modules import mutator
+
+from .common.utils import (
+    AnsibleExitJson, AnsibleFailJson, ModuleTestCase, set_module_args,
+)
+
+
+class TestMutator(ModuleTestCase):
+    def test_minimal_mutator_parameters(self, mocker):
+        sync_mock = mocker.patch.object(utils, 'sync')
+        sync_mock.return_value = True, {}
+        set_module_args(
+            name='test_mutator',
+            command='/bin/true',
+        )
+
+        with pytest.raises(AnsibleExitJson):
+            mutator.main()
+
+        state, _client, path, payload, check_mode = sync_mock.call_args[0]
+        assert state == 'present'
+        assert path == '/mutators/test_mutator'
+        assert payload == dict(
+            command='/bin/true',
+            metadata=dict(
+                name='test_mutator',
+                namespace='default',
+            ),
+        )
+        assert check_mode is False
+
+    def test_all_mutator_parameters(self, mocker):
+        sync_mock = mocker.patch.object(utils, 'sync')
+        sync_mock.return_value = True, {}
+        set_module_args(
+            name='test_mutator',
+            state='absent',
+            command='/bin/true',
+            timeout=30,
+            runtime_assets='awesomeness',
+            labels={'region': 'us-west-1'},
+            annotations={'playbook': 12345},
+        )
+
+        with pytest.raises(AnsibleExitJson):
+            mutator.main()
+
+        state, _client, path, payload, check_mode = sync_mock.call_args[0]
+        assert state == 'absent'
+        assert path == '/mutators/test_mutator'
+        assert payload == dict(
+            command='/bin/true',
+            timeout=30,
+            runtime_assets=['awesomeness'],
+            metadata=dict(
+                name='test_mutator',
+                namespace='default',
+                labels={'region': 'us-west-1'},
+                annotations={'playbook': '12345'},
+            ),
+        )
+        assert check_mode is False
+
+    def test_failure(self, mocker):
+        sync_mock = mocker.patch.object(utils, 'sync')
+        sync_mock.side_effect = errors.Error('Bad error')
+        set_module_args(
+            name='test_mutator',
+            command='/bion/true'
```

Comment:
Ahhhh this reminds me of the LEGO Bionicle. Ever played with [those](https://www.google.com/search?q=bionicle&client=firefox-b-d&tbm=isch&source=iu&ictx=1&fir=G2dj5JWKpZVJnM%253A%252CaV4NZ2PHs6O1XM%252C_&vet=1&usg=AI4_-kSsa3GpG1it-UCh_H7R3m8Iiezp9w&sa=X&ved=2ahUKEwik6qWK6ejkAhUIExoKHZ_RCGoQ9QEwA3oECAIQDw#imgrc=LVpzbLGS6fYHUM:&vet=1)? 😂 

Please don't fix the typo here, it's so amusing (and doesn't affect the test in any way) 👍

Reply:
Haha yeah I did, I still have them somewhere :laughing: